packages/a11y-base/src/aria-modal-controller.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright (c) 2021 - 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { hideOthers } from './aria-hidden.js';
+import { FocusTrapController } from './focus-trap-controller.js';
+
+/**
+ * A controller for handling modal state on the elements with `dialog` and `alertdialog` role.
+ * See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal
+ *
+ * Note, the actual `role` and `aria-modal` attributes are supposed to be handled by the
+ * consumer web component. This is done in to ensure the controller only does one thing.
+ */
+export class AriaModalController extends FocusTrapController {
```

Comment:
**suggestion:** Consider using composition over inheritance to avoid exposing unnecessary parts of the API.

```js
constructor(host) {
  this.focusTrapController = new FocusTrapController();
}
```

Reply:
Good point, thanks 👍

Reply:
Actually, I decided to decouple these controllers from each other. There's no reason to do otherwise.